### PR TITLE
Simhash gen

### PR DESCRIPTION
**Pull request by r4dix10**
Issue #12530 
@XVilka this is a very primitive implementation for generating hash for all features in a file.

Usage: rahash2 -qH <some-binary-file-path> > hash.txt
Ex: rahash2 -qH ../../hello > hello_hash.txt

Please Suggest some ways to filter only important features from binary.

Ex: When a person is generating simhash for ELF file. We can discard the features from the file which indicates that this is a ELF file.

### CODE REVIEW

File: binr/rahash2/rahash2.c [hunk 1]
```diff
@@ -469,9 +471,39 @@ int main(int argc, char **argv) {
...
+							fclose(fp);
+							return 0;
+						}
+						buf[i] = ch;
+					}
+					hashstr_hex = 0;
+					hashstr = buf;
+          goto simhash_label_forward;
+			} else {
+				printf ("No Such file exist");
```
Comment (radare) [inline binr/rahash2/rahash2.c]: errors/warnings go with eprintf, not printf

File: binr/rahash2/rahash2.c [hunk 2]
```diff
@@ -469,9 +471,39 @@ int main(int argc, char **argv) {
...
+			}
+		} else {
+			printf("No file path given\n");
+		}
+		return 0;
```
Comment (radare) [inline binr/rahash2/rahash2.c]: return 1

File: binr/rahash2/rahash2.c [hunk 3]
```diff
@@ -469,9 +471,39 @@ int main(int argc, char **argv) {
...
+	while (simhash_flag) {
+		FILE *fp;
+		if (argv[3]) {
+			int i, ch;
+			fp = fopen(argv[3], "r");
+			if (fp != NULL) {
+				char buf[8]; // shingle = 8
+				simhash_label_backward:
+					to = 0;
+					for (i = 0; i < 8; i++) {
```
[the remark this answers is not in the document]
Comment (r4dix10) [reply binr/rahash2/rahash2.c]: working on this.

File: binr/rahash2/rahash2.c [hunk 4]
```diff
@@ -496,12 +496,12 @@ int main(int argc, char **argv) {
 					hashstr = buf;
           goto simhash_label_forward;
 			} else {
-				printf ("No Such file exist");
+				eprintf ("No Such file exist");
 			}
 		} else {
-			printf("No file path given\n");
+			eprintf("No file path given\n");
```
Comment (radare) [inline binr/rahash2/rahash2.c]: missing space

**Pull request closed without merging**